Make minor edits to theme docs (#18297)
gatsbyjs/gatsby: docs/docs/themes/shadowing.md

### Applying new props

> In some cases components offer prop APIs to change their behavior. By extending this type of component you can import it and then add your new prop to change its behavior.

Reviewer: ```suggestion
In some cases components offer prop APIs to change their behavior. To extend a component you can import it and then add a new prop.
```
